api/novelist.py

```diff
@@ -551,31 +569,50 @@ def create_item_object(self, object, currentIdentifier, existingItem):
             # a data error.
             return (None, None, None, False)
 
-        role = object[5]
-        author = object[6] if role in Contributor.AUTHOR_ROLES else ""
+        roles = list(Contributor.AUTHOR_ROLES)
+        roles.append(Contributor.NARRATOR_ROLE)
+
+        role = object[6]
+        author_or_narrator = object[7] if role in roles else ""
```

Comment:
NoveList asked that narrator be provided separately from author, in a field called 'narrator'. So I think you need to keep track of the last author you saw _and_ the last narrator you saw.